tests/unittests/test_exception_handling.py

```diff
@@ -351,6 +458,222 @@ def test_check_forbidden_403_error_in_discovery_mode(self, mocked_refresh_creden
+
+        try:
+            xero_client.check_platform_access(config, config_path)
+        except client_.XeroNotAvailableError as e:
+            expected_message = "HTTP-error-code: 503, Error: API service is currently unavailable."
+            self.assertEqual(str(e), expected_message)
+
+
+    @mock.patch("tap_xero.client.XeroClient.refresh_credentials")
+    @mock.patch('requests.Request', side_effect=mocked_failed_429_request_in_day)
+    def test_too_many_requests_in_day_429_error_in_discovery_mode(self, mocked_refresh_credentials, mocked_session, mocked_failed_429_request_in_day):
+        
+        mocked_refresh_credentials.return_value = ""
+        config = {}
+        config_path = ""
+
+        xero_client = client_.XeroClient(config)
+        xero_client.access_token = "123"
+        xero_client.tenant_id = "123"
+
+        try:
+            xero_client.check_platform_access(config, config_path)
+        except client_.XeroTooManyError as e:
+            expected_message = "HTTP-error-code: 429, Error: The API rate limit for your organisation/application pairing has been exceeded. Please retry after 1000 seconds"
+            self.assertEqual(str(e), expected_message)
+
+
+    @mock.patch("tap_xero.client.XeroClient.refresh_credentials")
+    @mock.patch('requests.Request', side_effect=mocked_failed_429_request_in_minute)
+    def test_too_many_requests_in_minute_429_error_in_discovery_mode(self, mocked_refresh_credentials, mocked_session, mocked_failed_429_request_in_minute):
+
+        mocked_refresh_credentials.return_value = ""
+        config = {}
+        config_path = ""
+
+        xero_client = client_.XeroClient(config)
+        xero_client.access_token = "123"
+        xero_client.tenant_id = "123"
+
+        try:
+            xero_client.check_platform_access(config, config_path)
+        except client_.XeroTooManyInMinuteError as e:
+            expected_message = "HTTP-error-code: 429, Error: The API rate limit for your organisation/application pairing has been exceeded. Please retry after 5 seconds"
+            self.assertEqual(str(e), expected_message)
+
+
+    @mock.patch("tap_xero.client.XeroClient.refresh_credentials")
+    @mock.patch('requests.Request', side_effect=mocked_failed_429_request_in_day)
+    def test_too_many_requests_in_day_429_not_backoff_behavior_discovery_mode(self, mocked_refresh_credentials, mocked_session, mocked_failed_429_request_in_day):
+
+        mocked_refresh_credentials.return_value = ""
+        config = {}
+        config_path = ""
+
+        xero_client = client_.XeroClient(config)
+        xero_client.access_token = "123"
+        xero_client.tenant_id = "123"
+
+        try:
+            xero_client.check_platform_access(config, config_path)
+        except (requests.HTTPError, client_.XeroTooManyError) as e:
+            pass
+
+        #Verify daily limit should not backoff
+        self.assertEqual(mocked_failed_429_request_in_day.call_count, 1)
```

Comment:
@savan-chovatiya Can we have this assert in 'test_too_many_requests_429_in_day_error' function as well